WordPress/Classes/ViewRelated/Media/Tenor/TenorMedia.swift

```diff
@@ -0,0 +1,123 @@
+import MobileCoreServices
+import WPMediaPicker
+
+struct TenorImageCollection {
+    let largeURL: URL
+    let previewURL: URL
+    let staticThumbnailURL: URL
+    let largeSize: CGSize
+}
+
+// Models a Tenor image
+
+final class TenorMedia: NSObject {
+    let id: String
+    let name: String
+    let caption: String
+    let updatedDate: Date
+    let images: TenorImageCollection
+
+    init(id: String, name: String, caption: String, images: TenorImageCollection, date: Date? = nil) {
+        self.id = id
+        self.name = name
+        self.caption = caption
+        self.updatedDate = date ?? Date()
+        self.images = images
+    }
+}
+
+// MARK: - Create Tenor media from API GIF Entity
+
+extension TenorMedia {
+    convenience init?(tenorGIF gif: TenorGIF) {
+        let largeGif = gif.media.first { $0.gif != nil }?.gif
+        let previewGif = gif.media.first { $0.tinyGIF != nil }?.tinyGIF
+        let thumbnailGif = gif.media.first { $0.nanoGIF != nil }?.nanoGIF
+
+        guard largeGif != nil, previewGif != nil, thumbnailGif != nil else {
+            return nil
+        }
+
+        let images = TenorImageCollection(largeURL: largeGif!.url,
+                                          previewURL: previewGif!.url,
+                                          staticThumbnailURL: thumbnailGif!.url,
+                                          largeSize: largeGif!.mediaSize)
+
+        self.init(id: gif.id, name: gif.title ?? "", caption: "", images: images, date: gif.created)
```

Comment:
It appears there is never a Tenor `caption`. Should there be? If not, can this property be removed?

Reply:
Thanks @ScoutHarris, removed it ;-)